### ISSUE TITLE
Freezing the BatchNormalization layer during fine-tuning

### ISSUE THREAD

**Issue by jonbakerfish** [torch/nn#873]
What is the proper way to fix the parameters in [`BatchNormalization`](https://github.com/torch/nn/blob/master/BatchNormalization.lua) during fine-tuning (i.e. fix the mean(x), std(x), gamma and beta)? I noticed that there is a `self.train` variable control the behaviors of this layer. Is it all I need to do is to set it to false?

However, during training, we might call `model:tranining()`. This will set all the layer's `self.train` to true. In order to avoid modifying the training scripts, I'm thinking to add a variable `self.is_fixed` to the `BatchNormalization` layer and modify the [BN:updateOutput](https://github.com/torch/nn/blob/master/BatchNormalization.lua#L101) and [backward](https://github.com/torch/nn/blob/master/BatchNormalization.lua#L128) function by adding:

    if self.is_fixed then
        self.train = false
    end
    
Is there any better solution?

**Comment by soumith** [comment]
you can do this:

```lua
model:training()
model:apply(function(m) if torch.type(m):find("BatchNormalization") then m:evaluate() end end)
```

**Comment by fmassa** [comment]
@soumith what about adding a `freezeStats` function to `BatchNorm` ?

**Comment by soumith** [comment]
just adding a custom function to BatchNorm makes little sense

**Comment by LinZhineng** [comment]
hi @soumith I want to freeze the parameters of other layers, such as SpatialConvolution layer, whether the layer can be set to be "evaluate" mode. If not, any idea?

thanks and best regards!

**Comment by omair-kg** [comment]
@soumith 
If i want to freeze the weights of my network and just relearn the batch norm parameters, all i need to do is assign my network a LR of 0 and train again (for only one epoch), correct?
